File: LeaveTimesheetFrontend/LeaveTimesheetFrontend/src/app/core/models/employee.ts
export interface EmployeeDto {
  id: number;
  userId: number;
  employeeCode: string;
  fullName: string;
  email: string;
  phone?: string;
  departmentId?: number;
  departmentName?: string;
  managerId?: number;
  managerName?: string;
  designation?: string;
  dateOfJoining: string;
  isActive: boolean;
  gender?: string; 
}

export interface CreateEmployeeRequest {
  userId: number;         // links to auth user
  fullName: string;
  email: string;
  gender?: string;        // "Male" | "Female" | "Other"
  phone?: string;
  departmentId: number;
  managerId?: number | null;
  designation: string;
  employmentType?: string;
  dateOfJoining: string;
}

export interface DepartmentDto {
  id: number;
  name: string;
  code: string;
  managerId?: number;
    headId?: number; 
   headName?: string; 
  managerName?: string;
  isActive: boolean;
  employeeCount: number;
}

File: LeaveTimeSheetMS/EmployeeService/Domain/Entities/Employee.cs
namespace EmployeeService.Domain.Entities;

public class Employee
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string? Gender { get; set; }  // "Male", "Female", "Other"
    public string Email { get; set; } = string.Empty;
    public string EmployeeCode { get; set; } = string.Empty;
    public string Designation { get; set; } = string.Empty;
    public string EmploymentType { get; set; } = "Full-time";
    public int DepartmentId { get; set; }
    public Department Department { get; set; } = null!;
    public int? ManagerId { get; set; }
    public Employee? Manager { get; set; }
    public ICollection<Employee> DirectReports { get; set; } = new List<Employee>();
    public DateTime DateOfJoining { get; set; }
    public bool IsActive { get; set; } = true;
    public string? ProfilePhotoUrl { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}